docroot/modules/custom/sitenow_advanced_webform/src/Plugin/WebformHandler/ProspectorRemotePostWebformHandler.php

```diff
@@ -0,0 +1,134 @@
+
+  /**
+   * {@inheritdoc}
+   */
+  public function prepareForm(
+    WebformSubmissionInterface $webform_submission,
+    $operation,
+    FormStateInterface $form_state,
+  ) {
+    // @todo This doesn't seem quite right because it might show to an end user.
+    //   Maybe it should be limited by the user's permissions? Or maybe there is
+    //   a better place to show this message where only a form admin would see
+    //   it?
+    // Load configuration.
+    $config = $this->configFactory->get('sitenow_advanced_webform.settings');
+    $site_uuid = $config->get('prospector.site_uuid');
+    // We need a site UUID to proceed.
+    if (!$site_uuid) {
+      // Generate a URL for the sitenow_advanced_webform settings form.
+      $url = Url::fromRoute('sitenow_advanced_webform.settings_form')->toString();
+      // Print a message directing the user to the settings form.
+      $this->messenger()->addError($this->t('Prospector Site UUID is missing. Please visit the settings form at @url to configure.', [
+        '@url' => $url,
+      ]));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(WebformSubmissionInterface $webform_submission, $update = TRUE) {
+    // Load configuration.
+    $config = $this->configFactory->get('sitenow_advanced_webform.settings');
+    $site_uuid = $config->get('prospector.site_uuid');
+    // We need a site UUID to proceed.
+    if (!$site_uuid) {
+      // @todo Log this?
+      return;
+    }
+
+    // Load basic auth credentials from config.
+    $auth = $config->get('prospector.auth');
+    // We need auth credentials to proceed.
+    if (!isset($auth['user']) || !isset($auth['pass'])) {
+      // @todo I think this needs to be moved similar to the site_uuid check.
+      $this->messenger()->addError($this->t('Prospector authentication credentials are missing. Please contact the SiteNow team for assistance.'));
+      return;
+    }
+    $options = [
+      'auth' => array_values($auth),
+    ];
+
+    // Add data from first_name, last_name, email, and phone fields.
+    $elements = ['first_name', 'last_name', 'email', 'phone'];
+    $data = [];
+    foreach ($elements as $element) {
+      if ($value = $webform_submission->getElementData($element)) {
+        $data[$element] = $value;
+      }
+    }
+    $data['siteInteractionUuid'] = $site_uuid;
+    $options['json'] = $data;
+
+    // Send http request.
+    try {
+      // @todo Should the URL be stored in config or as a setting?
+      $response = $this->httpClient->request('POST', 'https://test.its.uiowa.edu/prospect-api/api/prospect/submit', $options);
+      // Prints the message received from the API.
+      $this->messenger()->addStatus($this->t('@response', [
+        '@response' => $response->getBody()->getContents(),
+      ]));
+    }
+    catch (GuzzleException $e) {
+      // Log the exception.
+      $this->getLogger()->error('An error occurred while posting the webform submission to Prospector. Error: @error', [
```

Comment:
should this also contain `$response->getBody()->getContents()`

Reply:
I'm not 100% sure, but I think this might be the same output as`$e->getMessage()` because of the type of exception.

Reply:
would a bad uuid be a way to test/confirm this?

Reply:
a bad uuid seems to still return a 200 OK and not trigger this

Reply:
Try commenting out line 113: `$options['json'] = $data;`

Reply:
That worked and does include the response.

<img width="1474" alt="Screenshot 2024-10-31 at 12 00 52 PM" src="https://github.com/user-attachments/assets/f14ed74f-6289-4f27-bd2b-731135f81479">